### PR TITLE
Enhance permissions on APIs

### LINKED ISSUE #3802: Ensure basic endpoints are accessible without authentication
### Event-related endpoints

- [x] Getting a list of events & details about a single event
- [x] Getting a list of sessions of an event and the details about a single session
- [x] Getting a list of microlocations of an event and the details about a single microlocation
- [x] Getting a list of tracks of an event and the details about a single track
- ~Getting a list of speakers of an event and the details about a single speaker~ [ Edited by Piyush: This need to be done separately since speakers API isn't ready yet ]
- [x] Getting a list of tickets of an event and the details about a single tickets
- [x] Getting call for speakers info of an event
- [x] Getting copyright info of an event

### Other endpoints
- [x] Getting settings (all fields except all client IDs, access tokens, secrets, passwords, email settings). But note, `stripe-publishable-key` is required.
- [x] Getting pages list & detail

### PR DESCRIPTION
**Pull request by poush**
Fixes #3802

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Removes the permission decorator from relationship end of events for GET requests and removed some redundant code.

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -420,7 +420,6 @@ def before_get_object(self, view_kwargs):
             event = self.session.query(Event).filter_by(identifier=view_kwargs['identifier']).one()
             view_kwargs['id'] = event.id
 
-    decorators = (jwt_required,)
```
Comment (SaptakS) [inline app/api/events.py]: Why should relationship have no decorators? Is this accessible by everyone? Won't that enable anyone to create a relationship between event and other apis?

**Comment by niranjan94** [comment]
@poush The base branch history was modified. Please update your branch and fix your commits.

**Comment by niranjan94** [comment]
> poush The base branch history was modified. Please update your branch and fix your commits.

@poush 👆

**Comment by poush** [comment]
@shubham-padia I believe it goes well with #4114

**Comment by magdalenesuo** [comment]
@shubham-padia Please help us close/merge this. Please approve if changes made per your request.

**Pull request merged**